### PR TITLE
Move Route validation into Route.clean method

### LINKED ISSUE #62: 500 when saving a Route with misconfigured parent / slug
When a route is saved in the admin with an invalid combination of `parent` and `slug`, a 500 is raised. This needs proper validation.

 | `parent` | no `parent`
--- | --- | ---
**`slug`** | branch / leaf node | *invalid*
**no `slug`** | *invalid* |  root node

screenshot from 2015-08-02 15-33-39

Perhaps the message could be clearer too?

### PR DESCRIPTION
**Pull request by Ian-Foote**
Fix #62.

### CODE REVIEW

File: conman/routes/tests/test_models.py [hunk 1]
```diff
@@ -60,10 +62,41 @@ def test_create_leaf_without_slug(self):
...
+
+    def test_validate_leaf_without_slug(self):
+        """Leaf Routes must have a slug."""
+        root_route = RootRouteFactory.create()
+        data = {'slug': '', 'parent': root_route.pk}
+
+        form = self.form(data)
+        form.is_valid()
+
+        self.assertIn('__all__', form.errors)
```
Comment (Ian-Foote) [inline conman/routes/tests/test_models.py]: This `form.errors` is:

```python
{'slug': ['This field is required.'], '__all__': ['Route can be a root, or have a slug, not both.']}
```

We can remove the `slug` error by adding `blank=True` to `Route.slug`, but I'm not sure if we need to.

Comment (Ian-Foote) [inline conman/routes/tests/test_models.py]: Ah, I just noticed you [do this](https://github.com/meshy/django-conman/blob/django-admin/conman/routes/models.py#L60) in #46.

Comment (meshy) [inline conman/routes/tests/test_models.py]: Perhaps best to add it here. I think that #46 is likely to be stalled for a while, tbh.

Comment (meshy) [inline conman/routes/tests/test_models.py]: FWIW, I think we need to -- it doesn't make sense for a route to have a `slug` if it's a root node.

File: conman/routes/models.py [hunk 2]
```diff
@@ -124,12 +134,7 @@ def save(self, *args, **kwargs):
 
         Adapted from feincms/module/page/models.py:248 in FeinCMS v1.9.5.
         """
-        is_root = self.parent_id is None
-        has_slug = bool(self.slug)
-
-        # Must be one or the other
-        if is_root == has_slug:
-            raise ValueError('Route can be a root, or have a slug, not both.')
+        self.clean()
```
Comment (meshy) [inline conman/routes/models.py]: Do the tests pass if you remove this?

**Comment by meshy** [comment]
Sorry, @Ian-Foote, I'm going to close this in favour of a new architecture that I hope will simplify things. (#75)

**Pull request closed without merging**